### PR TITLE
Export - navi data changes

### PR DESCRIPTION
**Pull request by ayeswarya**
Resolves #

## Description
Navi data changes for table export
## Proposed Changes

-

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/data/addon/adapters/facts/interface.ts [hunk 1]
```diff
@@ -116,17 +116,46 @@ export type AsyncQueryResponse = {
   };
 };
 
+export interface TableExport {
```
Comment (jkusa) [inline packages/data/addon/adapters/facts/interface.ts:119]: Is this type being used?

File: packages/data/addon/adapters/facts/elide.ts [hunk 2]
```diff
@@ -267,13 +268,80 @@ export default class ElideFactsAdapter extends EmberObject implements NaviFactAd
...
+    const tableExport = tableExportPayload?.tableExport.edges[0]?.node;
+    const { id } = tableExport;
+    let status: QueryStatus = tableExport.status;
+
+    while (status === QueryStatus.QUEUED || status === QueryStatus.PROCESSING) {
+      yield timeout(this._pollingInterval);
+      tableExportPayload = yield this.fetchTableExport(id, request.dataSource);
+      status = tableExportPayload?.tableExport.edges[0]?.node.status;
+    }
+    return tableExportPayload;
```
Comment (jkusa) [inline packages/data/addon/adapters/facts/elide.ts:332]: Can we add a finally case to handle the `cancel` test like we did for asyncQuery: https://github.com/yavin-dev/framework/blob/bafbc62e5ba699724194f88b79fd8d6b504d5fe3/packages/data/addon/adapters/facts/elide.ts#L367

File: packages/data/tests/unit/adapters/facts/elide-test.ts [hunk 3]
```diff
@@ -543,6 +585,127 @@ module('Unit | Adapter | facts/elide', function (hooks) {
     }
   });
 
+  test('urlForDownloadQuery - success', async function (assert) {
+    assert.expect(3);
+    const adapter: ElideFactsAdapter = this.owner.lookup('adapter:facts/elide');
+    const downloadURL = 'downloadURL';
```
Comment (jkusa) [inline packages/data/tests/unit/adapters/facts/elide-test.ts:591]:
```suggestion
    const downloadURL = 'https://yavin.dev/export/123';
```

File: packages/data/tests/unit/adapters/facts/elide-test.ts [hunk 4]
```diff
@@ -543,6 +585,127 @@ module('Unit | Adapter | facts/elide', function (hooks) {
...
+        },
+      };
+
+      return [500, { 'Content-Type': 'application/json' }, JSON.stringify({ data: response })];
+    });
+
+    try {
+      await taskFor(adapter.urlForDownloadQuery).perform(TestRequest, {});
+    } catch (e) {
+      console.log('Internal server error', e.errors[0].response.statusText);
```
Comment (jkusa) [inline packages/data/tests/unit/adapters/facts/elide-test.ts:660]:
```suggestion
```

File: packages/data/addon/adapters/facts/interface.ts (hunk 1 above)
Comment (ayeswarya) [reply packages/data/addon/adapters/facts/interface.ts:119]: Will remove AsyncQuery from here as well

File: packages/data/addon/adapters/facts/elide.ts [hunk 5]
```diff
@@ -267,13 +268,86 @@ export default class ElideFactsAdapter extends EmberObject implements NaviFactAd
...
+        yield this.cancelTableExport(id, request.dataSource);
+      }
+    }
   }
+
+  /**
+   * @param id
+   * @returns Promise that resolves to the result of the TableExport fetch query
+   */
+  fetchTableExport(id: string, dataSourceName?: string) {
```
Comment (jkusa) [inline packages/data/addon/adapters/facts/elide.ts:345]:
```suggestion
  fetchTableExport(id: string, dataSourceName: string) {
```

File: packages/data/addon/adapters/facts/elide.ts [hunk 6]
```diff
@@ -267,13 +268,86 @@ export default class ElideFactsAdapter extends EmberObject implements NaviFactAd
...
+    const query: DocumentNode = GQLQueries['tableExportFactsQuery'];
+    dataSourceName = dataSourceName || getDefaultDataSource().name;
```
Comment (jkusa) [inline packages/data/addon/adapters/facts/elide.ts:347]:
```suggestion
```

File: packages/data/addon/adapters/facts/elide.ts [hunk 7]
```diff
@@ -267,13 +268,86 @@ export default class ElideFactsAdapter extends EmberObject implements NaviFactAd
...
+      return tableExportPayload;
+    } finally {
+      if (status === QueryStatus.QUEUED || status === QueryStatus.PROCESSING) {
+        yield this.cancelTableExport(id, request.dataSource);
+      }
+    }
   }
+
+  /**
+   * @param id
```
Comment (jkusa) [inline packages/data/addon/adapters/facts/elide.ts:342]:
```suggestion
   * @param id
   * @param dataSourceName
```

File: packages/data/addon/adapters/facts/elide.ts [hunk 8]
```diff
@@ -267,13 +268,86 @@ export default class ElideFactsAdapter extends EmberObject implements NaviFactAd
...
+    // TODO: Add other options based on RequestOptions
+    const queryOptions = { mutation, variables: { id, query }, context: { dataSourceName } };
+    return this.apollo.mutate(queryOptions);
+  }
+
+  /**
+   * @param id
+   * @returns Promise with the updated tableExport's id and status
+   */
+  cancelTableExport(id: string, dataSourceName?: string) {
```
Comment (jkusa) [inline packages/data/addon/adapters/facts/elide.ts:290]:
```suggestion
  cancelTableExport(id: string, dataSourceName: string) {
```

File: packages/data/addon/adapters/facts/elide.ts [hunk 9]
```diff
@@ -267,13 +268,86 @@ export default class ElideFactsAdapter extends EmberObject implements NaviFactAd
...
+    const query = this.dataQueryFromRequest(request);
+    const id: string = options.requestId || v1();
+    const dataSourceName = request.dataSource || options.dataSourceName;
+    // TODO: Add other options based on RequestOptions
+    const queryOptions = { mutation, variables: { id, query }, context: { dataSourceName } };
+    return this.apollo.mutate(queryOptions);
+  }
+
+  /**
+   * @param id
```
Comment (jkusa) [inline packages/data/addon/adapters/facts/elide.ts:287]:
```suggestion
   * @param id
   * @param dataSourceName
```

File: packages/data/addon/adapters/facts/elide.ts [hunk 10]
```diff
@@ -267,13 +268,86 @@ export default class ElideFactsAdapter extends EmberObject implements NaviFactAd
...
+   * @returns Promise with the updated tableExport's id and status
+   */
+  cancelTableExport(id: string, dataSourceName?: string) {
+    const mutation: DocumentNode = GQLQueries['tableExportFactsCancel'];
+    dataSourceName = dataSourceName || getDefaultDataSource().name;
```
Comment (jkusa) [inline packages/data/addon/adapters/facts/elide.ts:292]:
```suggestion
```

**Review by jkusa — APPROVED** [review]
Small things, otherwise 👍

**Review by kevinhinterlong — APPROVED** [review]

**Pull request merged**